### PR TITLE
Feat/gradle

### PR DESCRIPTION
**Pull request by Kostiiii**
First step of introducing gradle to project.

Define tasks to build documentations, docker image, to run tests and some 'helpers' for installing requirements or project locally.

### CODE REVIEW

**Comment by Kostiiii** [comment]
I will update readme with how we can use gradle etc, but will wait with that until changes to readme structure.

File: .github/workflows/merge.yml [hunk 1]
```diff
@@ -22,38 +22,9 @@ jobs:
...
-
-      - name: "5. Run tests with coverage"
-        run: |
-          coverage run -m unittest
-          coverage report -m --fail-under=80
-
-      - name: "6. Run linter"
-        run: |
-          flake8 --config=setup.cfg darts
+          ./gradlew test
```
Comment (TheMP) [inline .github/workflows/merge.yml:27]: might still need the step to eg. attach the cache for pip, probably not a gradle thing

File: .github/workflows/merge.yml [hunk 2]
```diff
@@ -104,39 +56,10 @@ jobs:
         with:
           python-version: 3.6
 
-      - name: "3. Update pip and install pandoc"
+      - name: "3. Install pandoc"
```
Comment (TheMP) [inline .github/workflows/merge.yml:59]: Same remark - might still need to mount pip cache

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    daemonize false
+    clean true
+}
+
+// setup requirements
+task setupPip(type: Exec) {
+    commandLine "python", "-m", "pip", "install", "--upgrade", "pip"
+}
+
+void pipInstallSteps() {
```
Comment (TheMP) [inline build.gradle:52]: how can we pick some of those steps? some build tasks need main + release, some need main + dev so installing all of those is not needed

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        exec {
+            commandLine "jupyter", "nbextension", "enable", "--py", "widgetsnbextension"
+        }
+    }
+}
+
+// Documentation build
+void docSteps() {
+    exec {
+        commandLine "make", "--directory", "./docs", "copy-examples"
```
Comment (TheMP) [inline build.gradle:126]: is there any gradle plugin for Makefile?

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+            into "."
+        }
+        from(".") {
+            include "requirements/**"
+            into "."
+        }
+    }
+}
+
+dockerRun {
```
[the remark this answers is not in the document]
Comment (Kostiiii) [reply build.gradle:39]: scripts directory removed + readme updated.

File: .github/workflows/merge.yml (hunk 1 above)
Comment (Kostiiii) [reply .github/workflows/merge.yml:40]: Yes, sorry I went into berserk delete everything mode.

File: build.gradle (hunk 3 above)
Comment (Kostiiii) [reply build.gradle]: In that form we couldn't but now it is splitted into 3 task and in other places tasks now depend only on needed requirements.

File: build.gradle (hunk 4 above)
Comment (Kostiiii) [reply build.gradle:127]: Quick googling didn't revealed any trustworthy looking plugins and to be honest I feel like plugin for Makefile would be an overkill.

Comment (TheMP) [reply build.gradle:127]: Agreed

**Comment by TheMP** [comment]
> I will update readme with how we can use gradle etc, but will wait with that until changes to readme structure.

@Kostiiii new README is out so feel free to take a shot at it :)

**Comment by Kostiiii** [comment]
> > I will update readme with how we can use gradle etc, but will wait with that until changes to readme structure.
> 
> @Kostiiii new README is out so feel free to take a shot at it :)

@TheMP I think I extended README from develop branch already, did you see it? Do you want it to have some section I didn't include or to elaborate on something with more details?

**Pull request merged**